airbyte-metrics/lib/src/main/java/io/airbyte/metrics/lib/DogStatsDMetricSingleton.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.metrics.lib;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.timgroup.statsd.NonBlockingStatsDClientBuilder;
+import com.timgroup.statsd.StatsDClient;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Light wrapper around the DogsStatsD client to make using the client slightly more ergonomic.
+ * <p>
+ * This class mainly exists to help Airbyte instrument/debug application on Airbyte Cloud.
+ * <p>
+ * Open source users are free to turn this on and consume the same metrics.
+ */
+@Slf4j
+public class DogStatsDMetricSingleton {
+
+  private static DogStatsDMetricSingleton instance;
+  private final StatsDClient statsDClient;
+  private final boolean instancePublish;
+
+  public static synchronized DogStatsDMetricSingleton getInstance() {
+    if (instance == null) {
+      throw new RuntimeException("You must initialize configuration with the initialize() method before getting an instance.");
+    }
+    return instance;
+  }
+
+  public synchronized static void initialize(final AirbyteApplication app, final boolean publish) {
+    initialize(app, publish, null, null);
+  }
+
+  @VisibleForTesting
+  public synchronized static void initialize(final AirbyteApplication app, final boolean publish, final String ddAgentHost, final String ddPort) {
+    if (instance != null) {
+      throw new RuntimeException("You cannot initialize configuration more than once.");
+    }
+
+    if (!publish) {
+      log.info("Starting stub DogstatsD client..");
+      instance = new DogStatsDMetricSingleton();
+      return;
+    }
+
+    log.info("Starting DogStatsD client..");
+    if (!ddAgentHost.isBlank() || !ddPort.isBlank()) {
+      instance = new DogStatsDMetricSingleton(app.getApplicationName(), publish, ddAgentHost, ddPort);
+      return;
+    }
+
+    instance = new DogStatsDMetricSingleton(app.getApplicationName(), publish);
+  }
+
+  @VisibleForTesting
+  public synchronized static void flush() {
+    instance = null;
+  }
+
+  private DogStatsDMetricSingleton(final String appName, final boolean publish) {
+    instancePublish = publish;
+    statsDClient = new NonBlockingStatsDClientBuilder()
+        .prefix(appName)
+        .hostname(System.getenv("DD_AGENT_HOST")) // Matches Airbyte Cloud Datadog env vars.
+        .port(Integer.parseInt(System.getenv("DD_DOGSTATSD_PORT"))) // Matches Airbyte Cloud Datadog env vars.
+        .build();
```

Comment:
any reason not to use `this`?
```suggestion
    this(appName, publish, System.getenv("DD_AGENT_HOST"), Integer.parseInt(System.getenv("DD_DOGSTATSD_PORT"));
```

Reply:
I didn't realise that - nice